atak-civ/plugin-examples/plugintemplate/app/src/main/java/com/atakmap/android/plugintemplate/ActiveTimer.java

```diff
@@ -61,27 +91,105 @@ public ActiveTimerState getState() {
+            int importance = NotificationManager.IMPORTANCE_HIGH;
+            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
+            channel.setDescription(description);
+            channel.setShowBadge(false);
+            NotificationManager notificationManager = actualContext.getSystemService(NotificationManager.class);
+            notificationManager.createNotificationChannel(channel);
+        }
+    }
+
+    /**
+     * Creates (and updates a notification)
+     * Allows for user to pause or resume a timer using TimerNotificationActionReceiver
+     *
+     */
+    private void createNotification() {
+        Context actualContext = mapView.getContext();
+
+        Intent pauseIntent = new Intent("PAUSE");
+        pauseIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        pauseIntent.putExtra("activeTimer", notifyID);
+        PendingIntent pausePendingIntent =
+                PendingIntent.getBroadcast(actualContext, 0, pauseIntent, FLAG_UPDATE_CURRENT);
+        Intent resumeIntent = new Intent("RESUME");
+        pauseIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        resumeIntent.putExtra("activeTimer", notifyID);
+        PendingIntent resumePendingIntent =
+                PendingIntent.getBroadcast(actualContext, 0, resumeIntent, FLAG_UPDATE_CURRENT);
+        notificationStr = getDurationRemainingString();
+        NotificationCompat.Action pauseAction = new NotificationCompat.Action.Builder(android.R.drawable.ic_media_pause, "Pause", pausePendingIntent).build();
+        NotificationCompat.Action resumeAction = new NotificationCompat.Action.Builder(android.R.drawable.ic_media_play, "Resume", resumePendingIntent).build();
+
+        NotificationCompat.Builder builder;
+        if(state == ActiveTimerState.PAUSED) {
+            builder = new NotificationCompat.Builder(actualContext, CHANNEL_ID)
+                    .setSmallIcon(android.R.drawable.alert_light_frame)
+                    .setContentTitle(timer.getName())
+                    .setContentText(notificationStr)
+                    .setPriority(NotificationCompat.PRIORITY_LOW)
+                    .setColor(Color.blue(1))
+                    .setOnlyAlertOnce(true)
+                    .setSound(null)
+                    .addAction(resumeAction)
+                    .setOngoing(true);
+        } else {
+            builder = new NotificationCompat.Builder(actualContext, CHANNEL_ID)
+                    .setSmallIcon(android.R.drawable.alert_light_frame)
+                    .setContentTitle(timer.getName())
+                    .setContentText(notificationStr)
+                    .setPriority(NotificationCompat.PRIORITY_LOW)
+                    .setColor(Color.blue(1))
+                    .setOnlyAlertOnce(true)
+                    .setSound(null)
+                    .addAction(pauseAction)
+                    .setOngoing(true);
+        }
```

Comment:
Is it possible to add a reset function? I think that instead of giving the pause option when the timer is finished, it should give a reset option. You can check if the timer state is finished and then call the reset function in the ActiveTimer